junebug/plugins/nginx/plugin.py

```diff
@@ -0,0 +1,125 @@
+import logging
+import subprocess
+from os import path, remove
+from distutils.dir_util import mkpath
+from urlparse import urljoin
+
+from confmodel import Config
+from confmodel.fields import ConfigText
+
+from junebug.plugin import JunebugPlugin
+from junebug.utils import channel_public_http_properties
+
+
+log = logging.getLogger(__name__)
+DIRNAME = path.dirname(__file__)
+
+
+class NginxPluginConfig(Config):
+    '''Config for :class:`NginxJunebugPlugin`'''
+
+    vhost_file = ConfigText(
+        "The file to write the junebug nginx vhost file to",
+        default='/etc/nginx/sites-enabled/junebug.conf', static=True)
+
+    locations_dir = ConfigText(
+        "The directory to write location block config files to",
+        default='/etc/nginx/includes/junebug/', static=True)
+
+    server_name = ConfigText(
+        "Server name to use for nginx vhost",
+        required=True, static=True)
+
+    vhost_template = ConfigText(
+        "Path to the template file to use for the vhost config",
+        default='%s/vhost.template' % (DIRNAME,), static=True)
```

Comment:
Yep, this is for any vhost configuration people might want to do. I'm not sure if a text blob in a config field is a better way of approaching this though. A template of the file seems closer to the way this kind of thing is usually done with configuration management things like puppet. Any particular reason you think it would be easier?